dev/RadioMenuFlyoutItem/RadioMenuFlyoutItem.cpp

```diff
@@ -50,31 +67,51 @@ void RadioMenuFlyoutItem::OnInternalIsCheckedChanged(const winrt::DependencyObje
     else if (!IsChecked())
     {
         IsChecked(true);
-        UpdateSiblings();
+        UpdateCheckedItemInGroup();
     }
 }
 
-void RadioMenuFlyoutItem::UpdateSiblings()
+void RadioMenuFlyoutItem::UpdateCheckedItemInGroup()
 {
     if (IsChecked())
     {
-        // Since this item is checked, uncheck all siblings
-        if (auto parent = winrt::VisualTreeHelper::GetParent(*this))
+        const auto groupName = GroupName();
+
+        if (s_selectionMap.HasKey(groupName))
         {
-            const int childrenCount = winrt::VisualTreeHelper::GetChildrenCount(parent);
-            for (int i = 0; i < childrenCount; i++)
+            if (const auto previousCheckedItem = s_selectionMap.Lookup(groupName))
             {
-                auto child = winrt::VisualTreeHelper::GetChild(parent, i);
-                if (auto radioItem = child.try_as<winrt::RadioMenuFlyoutItem>())
+                // Uncheck the previously checked item.
+                previousCheckedItem.IsChecked(false);
+            }
+        }
+        s_selectionMap.Insert(groupName, *this);
+    }
+}
+
+void RadioMenuFlyoutItem::OnAreCheckStatesEnabledPropertyChanged(const winrt::DependencyObject& sender, const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    if (unbox_value<bool>(args.NewValue()))
+    {
+        if (auto const& subMenu = sender.try_as<winrt::MenuFlyoutSubItem>())
+        {
+            // Every time the submenu is loaded, see if it contains a checked RadioMenuFlyoutItem or not.
+            subMenu.Loaded(
```

Comment:
>subMenu.Loaded [](start = 12, length = 14)

This leaks the event registration. We should unhook this somewhere.

Reply:
fixed